lightning/src/routing/router.rs

```diff
@@ -925,7 +888,12 @@ where L::Target: Logger {
 				// fees caused by one expensive channel, but then this channel could have been used
 				// if the amount being transferred over this path is lower.
 				// We do this for now, but this is a subject for removal.
-				if let Some(available_value_contribution_msat) = available_liquidity_msat.checked_sub($next_hops_fee_msat) {
+				if let Some(mut available_value_contribution_msat) = htlc_maximum_msat.checked_sub($next_hops_fee_msat) {
+					let used_liquidity_msat = used_channel_liquidities
+						.get(&(short_channel_id, $src_node_id < $dest_node_id))
+						.map_or(0, |used_liquidity_msat| *used_liquidity_msat);
+					available_value_contribution_msat = available_value_contribution_msat
+						.saturating_sub(used_liquidity_msat);
```

Comment:
Is it ok that we continue after here when `available_value_contribution_msat` becomes zero?